src/view/forms/CaseFilterForm.tsx

```diff
@@ -2,42 +2,52 @@ import React from "react";
 import { RootState } from "../../redux/create";
 import { Dispatch, AnyAction, bindActionCreators } from "redux";
 import { casesActionCreators } from "../../redux/modules/cases";
-import { loadCases } from "../../redux/modules/casesAsync";
+import { caseFilterActionCreators } from "../../redux/modules/caseFilters";
+import { loadCases, loadSearchResults } from "../../redux/modules/casesAsync";
 import { connect } from "react-redux";
 import { CaseAgeFilter } from "./CaseAgeFilter";
 import { SnoozeReasonFilter } from "./SnoozeReasonFilter";
 import { ServiceNowFilter } from "./ServiceNowFilter";
+import { SnoozeStateFilter } from "./SnoozeStateFilter";
+import { CaseSearch } from "./CaseSearch";
 import { Well } from "../util/Well";
 import "./CaseFilterForm.scss";
+import UsaButton from "../util/UsaButton";
+import { hasFilters } from "../../redux/selectors";
 
 const mapStateToProps = (state: RootState) => ({
   caselist: state.cases.caselist,
-  start: state.cases.caseCreationStart,
-  end: state.cases.caseCreationEnd,
+  start: state.caseFilters.caseCreationStart,
+  end: state.caseFilters.caseCreationEnd,
   lastUpdated: state.cases.lastUpdated,
-  snoozeReasonFilter: state.cases.snoozeReasonFilter,
-  serviceNowFilter: state.cases.serviceNowFilter
+  snoozeReasonFilter: state.caseFilters.snoozeReasonFilter,
+  serviceNowFilter: state.caseFilters.serviceNowFilter,
+  snoozeState: state.caseFilters.snoozeState,
+  summary: state.cases.summary,
+  search: state.caseFilters.search,
+  activeSearch: state.caseFilters.activeSearch,
+  hasFilters: hasFilters(state)
 });
 
 const mapDispatchToProps = (dispatch: Dispatch<AnyAction>) =>
   bindActionCreators(
     {
       loadCases: loadCases,
+      loadSearchResults: loadSearchResults,
       clearCases: casesActionCreators.clearCases,
-      setStart: casesActionCreators.setCaseCreationStart,
-      setEnd: casesActionCreators.setCaseCreationEnd,
-      setSnoozeReasonFilter: casesActionCreators.setSnoozeReasonFilter,
-      setServiceNowFilter: casesActionCreators.setServiceNowFilter
+      setStart: caseFilterActionCreators.setCaseCreationStart,
+      setEnd: caseFilterActionCreators.setCaseCreationEnd,
+      setSnoozeReasonFilter: caseFilterActionCreators.setSnoozeReasonFilter,
+      setServiceNowFilter: caseFilterActionCreators.setServiceNowFilter,
+      setSnoozeState: caseFilterActionCreators.setSnoozeState,
```

Comment:
This seems annoying to maintain—is there any way to make the hand-wiring go away?

Reply:
I think the amount of boilerplate is a trade off you take when using a tool like Redux. We are using Redux to separate the "What happened" and the "How the state changed" from your display logic. I think the verbosity in this case ensures that we get maximum visibility into our application state changes. 

I think from a frontend standpoint the filters are an entire form not just a list of values. I think that with out clearly defining all the possible values for filters and seeing a clear repeating pattern then reducing filters to less then 1 field to 1 state value mapping would be of similar complexity to writing a generic form builder